packages/core/src/progress-circle/progress-circle.element.ts

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright (c) 2016-2020 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import {
+  baseStyles,
+  hasStringPropertyChanged,
+  property,
+  StatusTypes,
+  updateEquilateralSizeStyles,
+} from '@clr/core/internal';
+import { html, LitElement } from 'lit-element';
+import isNil from 'ramda/es/isNil';
+import { styles } from './progress-circle.element.css.js';
+
+/**
+ * Circular progress indicators provide a method to track how close long-running tasks are to
+ * completion. Circular progress offer a compact way to track progress in a variety of situations.
+ *
+ * ```typescript
+ * import '@clr/core/progress-circle/register.js';
+ * ```
+ *
+ * ```html
+ * <cds-progress-circle
+ *     status="info"
+ *     role="progressbar"
+ *     aria-valuenow="25"
+ *     aria-valuemin="0"
+ *     aria-valuemax="100"
+ *     value="25"></cds-progress-circle>
+ * ```
+ * @beta
+ * @element cds-progress-circle
+ * @cssprop --ringcolor
+ * @cssprop --fillcolor
+ */
+export class CdsProgressCircle extends LitElement {
+  private _size: string;
+
+  /**
+   * @type {default | info | success | warning | danger | unknown}
+   * Sets the color of the badge
+   */
+  @property({ type: String })
+  status: StatusTypes | 'unknown' = 'default';
+
+  /**
+   * Inverts color of circular progress bar if `true`.
+   * Useful for displaying icons on a dark background.
+   */
+  @property({ type: Boolean })
+  inverse = false;
+
+  /**
+   * Represents the percent completed from 0 to 100.
+   *
+   * If undefined, the progress-circle will be represented as an indeterminate
+   * progress indicator – a.k.a a "spinner".
+   */
+  @property({ type: Number })
+  value: number;
+
+  /**
+   * Represents the thickness of the stroke of the circular progress.
+   *
+   * If undefined, it defaults to 3.
+   */
+  @property({ type: Number })
+  line = 3;
+
+  private get radius() {
+    // 36 is the default viewbox dimensions; half of 36 is 18
+    // we need to keep this so that circular-progress remains aligned with our icons
+    // we shouldn't change it unless we change the default viewbox of the icons!
+    const halfOfViewbox = 18;
+
+    // line offset takes the width/thickness of the progress circle into account
+    const lineOffset = Math.ceil(this.line / 2);
+
+    return halfOfViewbox - lineOffset;
+  }
+
+  private get circumference() {
+    return 2 * Math.PI * this.radius;
+  }
+
+  /** @private */
+  get progress() {
```

Comment:
This is marked as private in the API but still reachable to make the unit tests easier to write.